streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -115,48 +121,45 @@ public void resume() {
      * 1. flush store
      * 2. write checkpoint file
      *
-     * @throws TaskMigratedException all the task has been migrated
      * @throws StreamsException fatal error, should close the thread
      */
     @Override
-    public void commit() {
-        switch (state()) {
-            case RUNNING:
-                stateMgr.flush();
-
-                // since there's no written offsets we can checkpoint with empty map,
-                // and the state current offset would be used to checkpoint
-                stateMgr.checkpoint(Collections.emptyMap());
-
-                offsetSnapshotSinceLastCommit = new HashMap<>(stateMgr.changelogOffsets());
-
-                log.info("Committed");
-                break;
-
-            case CLOSING:
-                // do nothing and also not throw
-                log.trace("Skip committing since task is closing");
-
-                break;
-
-            default:
-                throw new IllegalStateException("Illegal state " + state() + " while committing standby task " + id);
+    public void prepareCommit() {
```

Comment:
This is a meta comment: I think we can consolidate `prepareCommit` and `prepareClose` and `prepareSuspend` here by introducing the clean parameters to the function, since their logic are very similar (for the part that they diffs a bit, it can be pushed to post logic), and on task-manager during commit:

1) for each task -> task.prepareCommit(true)
2) commit
3) for each task -> task.postCommit(true)

During close:

if (clean)
    1) for each task -> task.prepareCommit(true)
    2) commit()
    3) for each task -> task.postCommit(true)
else
    1) for each task -> task.prepareCommit(false)
     // do not commit
    3) for each task -> task.postCommit(false)
4) tasks.close(flag)

And the same for suspension.

Reply:
Will do this in a follow up PR.